lib/go/test/flow_epoch_test.go

```diff
@@ -1516,3 +1517,555 @@ func TestEpochReset(t *testing.T) {
+			numClusters:          numClusters,
+			numEpochAccounts:     numEpochAccounts,
+			randomSource:         randomSource,
+			rewardIncreaseFactor: rewardIncreaseFactor,
+		}
+		runWithDefaultContracts(t, epochConfig, func(b emulator.Emulator, env templates.Environment, ids []string, idTableAddress flow.Address, IDTableSigner sdkcrypto.Signer, adapter *adapters.SDKAdapter) {
+			epochTimingConfigResult := executeScriptAndCheck(t, b, templates.GenerateGetEpochTimingConfigScript(env), nil)
+			var (
+				startView      uint64 = 100
+				stakingEndView uint64 = 120
+				endView        uint64 = 160
+				targetDuration uint64 = numEpochViews
+				// invalid epoch counter when recovering the current epoch the counter should equal the current epoch counter
+				epochCounter  uint64 = startEpochCounter + 100
+				targetEndTime uint64 = expectedTargetEndTime(epochTimingConfigResult, epochCounter)
+			)
+			args := getRecoveryTxArgs(env, ids, startView, stakingEndView, endView, targetDuration, targetEndTime, epochCounter)
+			// avoid using the recover epoch transaction template which has sanity checks that would prevent submitting the invalid epoch counter
+			code := `
+				import FlowEpoch from "FlowEpoch"
+				import FlowIDTableStaking from "FlowIDTableStaking"
+				import FlowClusterQC from "FlowClusterQC"
+				transaction(recoveryEpochCounter: UInt64,
+							startView: UInt64,
+							stakingEndView: UInt64,
+							endView: UInt64,
+							targetDuration: UInt64,
+							targetEndTime: UInt64,
+							clusterAssignments: [[String]],
+							clusterQCVoteData: [FlowClusterQC.ClusterQCVoteData],
+							dkgPubKeys: [String],
+							nodeIDs: [String],
+							unsafeAllowOverwrite: Bool) {
+
+					prepare(signer: auth(BorrowValue) &Account) {
+						let epochAdmin = signer.storage.borrow<&FlowEpoch.Admin>(from: FlowEpoch.adminStoragePath)
+							?? panic("Could not borrow epoch admin from storage path")
+
+						epochAdmin.recoverNewEpoch(
+								recoveryEpochCounter: recoveryEpochCounter,
+								startView: startView,
+								stakingEndView: stakingEndView,
+								endView: endView,
+								targetDuration: targetDuration,
+								targetEndTime: targetEndTime,
+								clusterAssignments: clusterAssignments,
+								clusterQCVoteData: clusterQCVoteData,
+								dkgPubKeys: dkgPubKeys,
+								nodeIDs: nodeIDs
+							)
+					}
+				}
+			`
+			tx := createTxWithTemplateAndAuthorizer(b, []byte(templates.ReplaceAddresses(code, env)), idTableAddress)
+			for _, arg := range args {
+				tx.AddArgument(arg)
+			}
+
+			expectedErr := fmt.Errorf("recovery epoch counter should equal current epoch counter + 1")
+			assertTransactionReverts(
+				t, b, tx,
+				[]flow.Address{idTableAddress},
+				[]sdkcrypto.Signer{IDTableSigner},
+				expectedErr,
+			)
+		})
+	})
+
+	t.Run("Recover epoch transaction panics when recovery epoch counter is less than currentCounter and unsafeAllowOverwrite is false", func(t *testing.T) {
```

Comment:
Regarding the set of test cases about when we expect the transaction to panic, I feel like they can be structured to more clearly communicate our expectations. 

The expectations are simple:
- if you are recovering by overwriting the current epoch, you must pass in exactly `recoveryEpochCounter = currentEpochCounter` and `unsafeAllowOverwrite = true`
- if you are recovering by transitioning into a new epoch, you must pass in exactly `recoveryEpochCounter = currentEpochCounter+1` and `unsafeAllowOverwrite = true`
- every other possible combination will panic

These expectations are communicated well by the descriptions of the first two test cases:

(1)
> 	t.Run("Panics when recovering a new epoch and recover epoch counter is not equal to the current epoch counter + 1", func(t *testing.T) {

(2)
> 	t.Run("Panics when recovering the current epoch and recover epoch counter is not equal to the current epoch counter", func(t *testing.T) {

The subsequent two test cases add more coverage by testing more input values (👍), but their descriptions describe subsets of the expectations described by (1) and (2). For me at least this made it harder to understand as a whole. I think the tests would be clearer if we added the invalid inputs we want to test within those two first test cases, where we fully define the range of valid input values that we are testing. 

Suppose $C$ is the `currentEpochCounter` value. Then, for example, (1) could test `recoveryEpochCounter` inputs C, C-1, C+2. (2) could test `recoveryEpochCounter` inputs C-1, C+1.

For each of (1) and (2), we can extract the test functionality into a closure and then do something like:
```go
invalidEpochCounters := []uint64{C, C-1, C+2} // C+1 is the only valid input
for _, epochCounter := range invalidEpochCounters {
  // run the test case
}
```